File: 06mostFrequentChar/mostFrequentChar.js
// My Solution
const mostFrequentChar = (s) => {
  
  const count = {}
  
  for (let char of s){
    if(!(char in count)){
      count[char] = 1
    }
    count[char] += 1
  }
 
  let values = Object.values(count)
  values.sort().reverse()
  for(key in count){
    if(count[key]===values[0])
      {return key}
  }
  
};

mostFrequentChar('bookeeper');

// Alvin's Solution

const mostFrequentChar = (s) => {
  const count = {};
  
  for (let char of s) {
    if (!(char in count)) {
      count[char] = 0;
    }
    count[char] += 1
  }
  
  let best = null;
  for (let char of s) {
    if (best === null || count[char] > count[best]) {
      best = char;
    }
  }
  return best;
};

File: Structy_1/02_Array_and_String/06_mostFrequentChar/most_frequent_char.py
from collections import Counter

def most_frequent_char(s):
#   Make counter object
  obj = Counter(s)
  print(obj)
  #   Make list of all values
  v=list(obj.values())
  print(v)
#   Make list of all keys
  k=list(obj.keys())
  print(k)
# return the first value of the list of keys
# at the index of the highest number
  return k[v.index(max(v))]

  # Alvin's Solution

  def most_frequent_char(s):
    count = {}
  for char in s:
    if char not in count:
      count[char] = 0    
    count[char] += 1
    
  best = None
  for char in s:
    if best is None or count[char] > count[best]:
      best = char
  return best